src/bridges/notional/NotionalBridge.sol

```diff
@@ -0,0 +1,207 @@
+     * we need to call mintViaAsset. Otherwise, we just need to call mintViaUnderlying.
+     * @dev Will revert if the wrapped fcash contract has not been deployed
+     * @param _inputToken The address of the input asset
+     * @param _outputToken The address of the output asset
+     * @param _currencyId The predefined Notional currency id defined in constructor
+     * @param _amount The amount of input token
+     * @param _maturity The maturity of the market we want to enter
+     * @return outputAmount The amount of fcash token we actually receive
+     */
+    function _enter(
+        address _inputToken,
+        address _outputToken,
+        uint16 _currencyId,
+        uint256 _amount,
+        uint40 _maturity
+    ) internal returns (uint256) {
+        IWrappedfCash fcash = IWrappedfCash(FCASH_FACTORY.computeAddress(_currencyId, _maturity));
+        if (address(fcash) != _outputToken) {
+            revert ErrorLib.InvalidOutputA();
+        }
+        if (_inputToken == address(0)) {
+            WETH9(WETH).deposit{value: _amount}();
+            _inputToken = WETH;
+        }
+        if (!preapproved[_inputToken][address(fcash)]) {
+            ERC20(_inputToken).approve(address(fcash), type(uint256).max);
+            preapproved[_inputToken][address(fcash)] = true;
+        }
+        if (fcash.getCurrencyId() != _currencyId) {
+            revert("fcash has not been deployed");
+        }
+        (IERC20 underlyingToken, ) = fcash.getUnderlyingToken();
+        if (address(underlyingToken) == ETH) {
+            underlyingToken = IERC20(WETH);
+        }
+        uint88 fcashAmount = computeUnderlyingAmount(_inputToken, address(underlyingToken), _amount);
+        if (address(underlyingToken) != _inputToken) {
+            fcash.mintViaAsset(_amount, fcashAmount, address(this), 0);
+        } else {
+            fcash.mintViaUnderlying(_amount, fcashAmount, address(this), 0);
+        }
+        return ERC20(address(fcash)).balanceOf(address(this));
+    }
+
+    /**
+     * @notice Redeem fcash to the underlying token or the asset token
+     * @dev Call redeemToAsset or redeemToUnderlying depends on the type of input token
+     * @dev Since there might be a slippage, we need to compute the number
+     * of received tokens manually
+     * @param _inputToken The address of the input asset
+     * @param _outputToken The address of the output asset
+     * @param _amount The amount of the input asset
+     * @return outputAmount The actual received amount of the output asset
+     */
+    function _exit(
+        address _inputToken,
+        address _outputToken,
+        uint256 _amount
+    ) internal returns (uint256) {
+        bool isETH = address(_outputToken) == address(0);
+        uint256 prevBalance = isETH
+            ? IERC20(WETH).balanceOf(address(this))
+            : IERC20(_outputToken).balanceOf(address(this));
```

Comment:
The bridge is not expected to hold any tokens between interaction so it would be safe to just return full balance to the users. If someone send some tokens to the bridge randomly, those users will be lucky. For this reason please remove the prevBalance variable.